Fallback on viewbox attribute

guillomovitch opened the pull request:
Some SVG file don't have size/heigth attributes, but only a viewbox attribute, such as this one:
https://rendez-vous.renater.fr/home/assets/images/logo.svg

ImageMagick does use those values to compute image dimensions, the following PR enforces the same behaviour.

guillomovitch committed:
fallback on viewBox if needed

Use viewbox attribute if width or heigth attribute is missing

guillomovitch committed:
fix splitting on alternative

guillomovitch committed:
round to integer value

guillomovitch committed:
same behaviour with XML::Simple

eserte reviewed (changes requested):
Can you check why the tests start to fail with your changes?

And can you provide additional test cases for your use case? Maybe with a small (!) sample svg file?

guillomovitch committed:
a bit more readability

guillomovitch committed:
test fallback behaviour

guillomovitch commented:
I just added a few tests, and the test suite runs fine for me.

guillomovitch committed:
make 'no' the default value for XML standalone property

That mimic XML::LibXML behaviour, and seems the default value according
to the specification anyway

guillomovitch commented:
Test results are not available anymore, and I don't have any access to the failing platform. Is there any way to rerun those tests ?
